### ISSUE TITLE
Could not find package in your path

### ISSUE THREAD

**Issue by Kraymer** [seddonym/layer_linter#10]
* Layer Linter version: 0.4.0 (but `layer-lint --version` does not work)
* Python version: 2.7.12
* Operating System: Ubuntu 16.04

### Description

Running the program

### What I Did

```
$ layer-lint test
Could not find package 'test' in your path.
$ echo $PYTHONPATH 
/home/flap/Dev/jlmcc/apps
$ pwd
/home/flap/Dev/jlmcc/apps
$ ls test
__init__.py  [...]
$ more layers.html 
rental:
    packages:
        - test
        - test/rental
    layers:
        - core
        - domain
```
I expect to have more infos in the doc to make this first `__import__` works

**Comment by seddonym** [comment]
Thanks for the bug report @Kraymer. I'm afraid Layer Linter doesn't support Python 2; it's possibly a consequence of that.

Are you able to reproduce the issue with Python 3?

**Comment by seddonym** [comment]
P.S. Good suggestion about `--version` flag, I've added it as an issue: https://github.com/seddonym/layer_linter/issues/11

**Comment by Kraymer** [comment]
PS: trying to make it work on a django project

**Comment by seddonym** [comment]
Layer Linter looks at packages in the Python path, rather than in the filesystem. So you could in theory run it on any package that is installed and importable from your Python prompt. In this case it's linting the built in test module.

Could you try running the Python prompt and then typing `import myproject`?

One other thing: your layers file should be called `layers.yml` (not html) and the packages should be listed as Python packages, not files. For example, if you wanted core and domain layers within two subpackages in your project, it would look something like this:

```
rental:
    packages:
        - myproject.subpackage1
        - myproject.subpackage2
    layers:
        - core
        - domain
```

Does that make sense?

**Comment by Kraymer** [comment]
```
╰─ python3 
Python 3.5.2 (default, Nov 23 2017, 16:37:01) 
[GCC 5.4.0 20160609] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import apps
>>> 
 
╰─ more layers.yml 
rental:
    packages:
        - apps.test
        - apps.test.rental
    layers:
        - core
        - domain

╰─ python3 ~/.local/bin/layer-lint apps
Could not find package 'apps' in your path.
```

The import works ok in the interpreter.

**Comment by Kraymer** [comment]
I can somehow make it work if I add a breakpoint in the code *(sigh)*

```                                                                                                                                                                                 
╰─ layer-lint apps
Could not find package 'apps' in your path.

// add breakpoint

╰─ layer-lint apps
> /home/flap/Dev/jlmcc/lib/python2.7/site-packages/layer_linter/cmdline.py(50)_main()
     49         import ipdb;ipdb.set_trace()
---> 50         package = __import__(package_name)
     51     except ImportError:

ipdb> n
> /home/flap/Dev/jlmcc/lib/python2.7/site-packages/layer_linter/cmdline.py(54)_main()
     53 
---> 54     if config_directory is None:
     55         config_directory = os.getcwd()

ipdb> c
============
Layer Linter
============

---------
Contracts
---------

rental KEPT

Contracts: 1 kept, 0 broken.
```

**Comment by seddonym** [comment]
Hmm, I'm not sure what's going on here and it's difficult to troubleshoot without your package / setup. 

Try doing this:

1. Create a Python3 virtualenv, and install your project's requirements in it, plus layer-linter.
2. Change directory to your project's root, and create your `layers.yml` in that same directory.
3. Run `layer-lint yourproject`.

If it still doesn't work, you could try removing the exception handling for the ImportError in `layer_linter/cmdline.py` and see if there's a more helpful exception raised.

Alternatively if you're able to share with me with an example project that triggers the error, I could see if I can reproduce it.

Thanks so much for your patience!

**Comment by seddonym** [comment]
Hi @Kraymer, I've just released an update (0.5.0) which fixes some problems with importing certain packages. It may fix your problem, if you would like to try it.

If you still have issues, could you run the program including the `--debug` flag, and paste the output here?

**Comment by seddonym** [comment]
I'm going to close this issue, as I believe it will have been addressed, if not in earlier updates, by the rewrite of the static analysis in the latest version. Feel free to open a new bug report if you still find you have issues.

Thanks again for the bug report.
